docs/src/testing/contract-collection/new-mechanism.md

```diff
@@ -0,0 +1,36 @@
+# How Contracts Are Collected
+
+For the [`declare`](../../appendix/snforge-library/declare.md) to function, snforge must collect and call build on
+contracts in the package. By default, if using Scarb version greater or equal to 2.8.3, snforge will combine test
+collection and contract collection steps.
+
+When running `snforge test`, snforge will, under the hood, call the `scarb build --test` command. This command builds
+all the test and contracts along them. Snforge collects these contracts and makes them available for declaring in tests.
+
+Contracts are collected from both `src` and `tests` directory, including modules marked with `#[cfg(test)]`.
+Internally, snforge collects contracts from all `[[test]]` targets compiled by Scarb.
+You can read more about that in [test collection](../test-collection.md) documentation.
```

Comment:
Not clear what this sentence means. It suggests that you can read more about collecting contracts in the test collection documentation, which is not the case.

Reply:
I meant that test collection docs contain more details on `[[test]]` target. 
How about this?

```suggestion
You can read more about the `[[test]]` target in [test collection](../test-collection.md) documentation.
```

Reply:
Sorry, I reviewed this before I saw the PR with the updated test collection docs. Everything makes sense now as it is.